sql-tuning-best-practice.md

```diff
@@ -0,0 +1,984 @@
+- [`tidb_auto_analyze_start_time`](/system-variables.md#tidb_auto_analyze_start_time) and [`tidb_auto_analyze_end_time`](/system-variables.md#tidb_auto_analyze_end_time): the time window for automatic statistics collection.
+
+```sql
+SHOW VARIABLES LIKE 'tidb\_auto\_analyze%';
+```
+
+```
++-----------------------------------------+-------------+
+| Variable_name                           | Value       |
++-----------------------------------------+-------------+
+| tidb_auto_analyze_ratio                 | 0.5         |
+| tidb_auto_analyze_start_time            | 00:00 +0000 |
+| tidb_auto_analyze_end_time              | 23:59 +0000 |
++-----------------------------------------+-------------+
+```
+
+In some cases, automatic collection might not meet your needs. By default, it occurs between `00:00` and `23:59`, meaning the analyze job can run at any time. To minimize performance impact on your online business, you can set specific start and end times for statistics collection.
+
+You can manually collect statistics using the `ANALYZE TABLE table_name` statement. This lets you adjust settings such as the sample rate, the number of Top-N values, or collect statistics for specific columns only.
+
+Note that after manual collection, subsequent automatic collection jobs inherit the new settings. This means that any customizations made during manual collection will apply to future automatic analyses.
+
+Locking table statistics is useful in the following scenarios:
+
+- The statistics on the table already represent the data well.
+- The table is very large, and statistics collection is time-consuming.
+- You want to maintain statistics only during specific time windows.
+
+To lock statistics for a table, you can use the [`LOCK STATS table_name`](/sql-statements/sql-statement-lock-stats.md) statement.
+
+For more information, see [Statistics](/statistics.md).
+
+### Understand execution plans
+
+An execution plan details the steps that TiDB will follow to execute a SQL query. This section explains how TiDB builds an execution plan and how to generate, display, and interpret execution plans.
+
+#### How TiDB builds an execution plan
+
+A SQL statement undergoes three main optimization stages in the TiDB optimizer:
+
+1. [Pre-processing](#1-pre-processing)
+2. [Logical transformation](#2-logical-transformation)
+3. [Cost-based optimization](#3-cost-based-optimization)
+
+##### 1. Pre-processing
+
+During pre-processing, TiDB determines whether the SQL statement can be executed using [`Point_Get`](/explain-indexes#point_get-and-batch_point_get) or [`Batch_Point_Get`](/explain-indexes#point_get-and-batch_point_get). These operations use a primary or unique key to read directly from TiKV through an exact key lookup. If a plan qualifies for `Point_Get` or `Batch_Point_Get`, the optimizer skips the logical transformation and cost-based optimization steps because direct key lookup is the most efficient way to access the row.
```

Comment:
```suggestion
During pre-processing, TiDB determines whether the SQL statement can be executed using [`Point_Get`](/explain-indexes.md#point_get-and-batch_point_get) or [`Batch_Point_Get`](/explain-indexes.md#point_get-and-batch_point_get). These operations use a primary or unique key to read directly from TiKV through an exact key lookup. If a plan qualifies for `Point_Get` or `Batch_Point_Get`, the optimizer skips the logical transformation and cost-based optimization steps because direct key lookup is the most efficient way to access the row.
```